### PR TITLE
docs(fargate): Documenting persistence for logs and events

### PR DESCRIPTION
**Pull request by rnishtala-sumo**
Fargate documentation updates
- Setting up persistence for logs metrics and events
- Document restructuring

### Checklist

- [x] Documentation updated

### CODE REVIEW

File: docs/fargate.md [hunk 1]
```diff
@@ -471,30 +548,187 @@ spec:
...
 kubectl apply -f sumo-metrics-pvc.yaml -n "${NAMESPACE}"
 ```
 
-### Install or upgrade collection
+#### Create `sumo-logs-pvc.yaml` with PVC per access point
 
-Disable Prometheus Node Exporter. As Amazon Fargate does not support Daemonsets, they won't be working correctly. There is no need to have
-Node Exporter on Amazon Fargate as Amazon takes care of Nodes management. To disable Prometheus Node Exporter, please add the following
-configuration to `user-values.yaml`:
+The objects can be created using the following `bash` script:
```
Comment (swiatekm-sumo) [inline docs/fargate.md:567]: Do you actually need to do all this? Can't you just create the StorageClass and then set `metadata.persistence.storageClass` to the name?

File: docs/fargate.md [hunk 2]
```diff
@@ -2,38 +2,42 @@
 
 **NOTE: This is an experimental feature and it's not a subject of breaking changes policy.**
 
+The following are some limitations of deploying this helm chart on EKS fargate
+
+- Only supports EKS version 1.24 and above
+- Does not support multiline logs
```
[the remark this answers is not in the document]
Comment (aboguszewski-sumo) [reply docs/fargate.md:8]: Sounds fair.

File: docs/fargate.md (hunk 1 above)
Comment (rnishtala-sumo) [reply docs/fargate.md:567]: yes, those persistence volumes and PVCs need to be created manually, this is the tedious prerequisite of deploying the helm chart with persistence enabled
https://aws.amazon.com/blogs/aws/new-aws-fargate-for-amazon-eks-now-supports-amazon-efs/

File: docs/fargate.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (rnishtala-sumo) [reply docs/fargate.md:8]: almost got it to work with logstransform

```
logstransform/cloudwatch:
    operators:
      - id: merge-cri-lines
        combine_field: attributes.log
        combine_with: ""
        is_last_entry: attributes.logtag == "F"
        output: "merge-multiline-logs"
        overwrite_with: newest
        source_identifier: attributes["cloudwatch.log.stream"]
        type: recombine
      - id: merge-multiline-logs
        combine_field: attributes.log
        combine_with: "\n"
        is_first_entry: attributes.log matches {{ .Values.sumologic.logs.multiline.first_line_regex | quote }}
        source_identifier: attributes["cloudwatch.log.stream"]
        type: recombine
      - field: attributes["cloudwatch.log.stream"]
        type: remove
```
Sample logs
https://stagdata.long.sumologic.net/ui/#/search/create?id=VyRCPomxbWEOky6av1FtXPwut6pVUqJ7UChca2tw

Will look at this issue for the next version.

File: docs/fargate.md (hunk 1 above)
Comment (swiatekm-sumo) [reply docs/fargate.md:567]: Ok, dynamic provisioning is not currently supported on Fargate, we can't do any better.

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**